File: Core/Entities/Concrete/Log.cs
namespace Core.Entities.Concrete;

public class Log : IEntity, ITenancy
{
    public int Id { get; set; }
    public int TenantId { get; set; }
    public string MessageTemplate { get; set; }
    public string Level { get; set; }
    public DateTime TimeStamp { get; set; }
    public string Exception { get; set; }
}

File: UiPreparation/FlutterUi/flutter_devarchitecture/lib/core/features/admin_panel/log_f/models/log.dart
import '../../../../models/i_entity.dart';

class Log implements IEntity {
  @override
  int id;
  String messageTemplate;
  String level;
  DateTime timeStamp;
  String exception;

  Log({
    required this.id,
    required this.messageTemplate,
    required this.level,
    required this.timeStamp,
    required this.exception,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'messageTemplate': messageTemplate,
      'level': level,
      'timeStamp': timeStamp,
      'exception': exception,
    };
  }

  factory Log.fromMap(Map<String, dynamic> map) {
    return Log(
      id: map['id'],
      messageTemplate: map['messageTemplate'],
      level: map['level'],
      timeStamp: map['timeStamp'],
      exception: map['exception'],
    );
  }
}
